### PR TITLE
Added forwarded headers options to startup

### PR DESCRIPTION
**Pull request by patrikulus**
Authentication with external providers stopped working some time ago. 
TL/DR: Forwarded Headers Middleware must be enabled when application is behind reverse proxy.

## Description
DevAdventCalendar is hosted behind traefik. It means, that requests are coming into app from another container, not directly. It seems that ASP.NET Core 2.2 is checking origin when it comes to OAuth authentication. So to get it work the right way, we need to forward origin url and protocol of the request. This can be done easily with middleware. For details, please check this [link](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/?view=aspnetcore-2.2&tabs=visual-studio#forward-request-information-with-a-proxy-or-load-balancer).

## Where should the reviewer start?
This is very specific kind of change which can be tested only on uat or production env (because OAuth providers have configured redirect urls and only those are valid).

## Motivation and context
The problem with login showed up in 2019 version. It must have been some nuget update, because it still works in 2018 version.

## How has this been tested?
On dev/uat environment with facebook provider.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by WTobor** [comment]
Browser Brave, register by twitter :(

image

**Comment by patrikulus** [comment]
Fixed it, and it covers #203 too ;)

**Comment by patrikulus** [comment]
Tested on dev/uat with success 😄
